client/keys.go

```diff
@@ -313,7 +349,11 @@ func (k *Key) Unseal(in *pb.SealedBytes, opts CertifyOpts) ([]byte, error) {
 	}
 	defer tpm2.FlushContext(k.rw, sealed)
 
-	if opts != nil {
+	pcrs, err := mergePCRSelAndProto(k.rw, opts.CertifyCurrent, opts.CertifyExpected)
+	if err != nil {
+		return nil, fmt.Errorf("invalid UnsealOpts: %v", err)
+	}
+	if len(pcrs.GetPcrs()) > 0 {
```

Comment:
Should we move the `CheckSubset` to after this line. That way we fail early on bad `UnsealOpts`. For example, if a user uses a hash algorithm in `UnsealOpts` that's not `CertifyHashAlgTpm`, we want to get the error
```
failed to certify PCRs: PCR hash algo not matching: ...
```
sooner rather than even bothering with the process of certification.

Reply:
this makes sense to me! nice call.